tests/study_tests/test_dataframe.py

```diff
@@ -159,27 +159,35 @@ def f(trial: Trial) -> float:
     ],
 )
 @pytest.mark.parametrize("multi_index", [True, False])
+@pytest.mark.parametrize("metric_names", [None, ["v0", "v1"]])
 def test_trials_dataframe_with_multi_objective_optimization(
-    attrs: Tuple[str, ...], multi_index: bool
+    attrs: tuple[str, ...], multi_index: bool, metric_names: list[str] | None
 ) -> None:
-    def f(trial: Trial) -> Tuple[float, float]:
+    def f(trial: Trial) -> tuple[float, float]:
         x = trial.suggest_float("x", 1, 1)
         y = trial.suggest_float("y", 2, 2)
 
         return x + y, x**2 + y**2  # 3, 5
 
     study = create_study(directions=["minimize", "maximize"])
+    if metric_names is not None:
+        study.set_metric_names(metric_names)
     study.optimize(f, n_trials=3)
     df = study.trials_dataframe(attrs=attrs, multi_index=multi_index)
 
-    if multi_index:
-        for i in range(3):
-            assert df.get("values")[0][i] == 3
-            assert df.get("values")[1][i] == 5
-    else:
-        for i in range(3):
-            assert df.values_0[i] == 3
-            assert df.values_1[i] == 5
+    def _get_value(i: int, j: int) -> float | None:
+        if multi_index and metric_names is not None:
+            return df.get("values")[metric_names[i]][j]
+        elif multi_index and metric_names is None:
+            return df.get("values")[i][j]
+        elif not multi_index and metric_names is not None:
+            return df.get(f"values_{metric_names[i]}")[j]
+        else:
+            return df.get(f"values_{i}")[j]
+
+    for i, v in zip(range(2), [3, 5]):
+        for j in range(3):
+            assert _get_value(i, j) == v
```

Comment:
I still see complex conditional test logic in this test case. As described in xUnit Test Patterns, it can be difficult to determine whether the problem is with the implementation or with the test itself when the test fails.
http://xunitpatterns.com/Conditional%20Test%20Logic.html

How about rewriting the test case as follows?

```python
@pytest.mark.parametrize("attrs", [("value",), ("values",)])
@pytest.mark.parametrize("multi_index", [True, False])
def test_trials_dataframe_with_multi_objective_optimization(
    attrs: tuple[str, ...], multi_index: bool
) -> None:
    def f(trial: Trial) -> tuple[float, float]:
        x = trial.suggest_float("x", 1, 1)
        y = trial.suggest_float("y", 2, 2)

        return x + y, x**2 + y**2  # 3, 5

    # without set_metric_names
    study = create_study(directions=["minimize", "maximize"])
    study.optimize(f, n_trials=1)
    df = study.trials_dataframe(attrs=attrs, multi_index=multi_index)
    if multi_index:
        assert df.get("values")[0][0] == 3
        assert df.get("values")[1][0] == 5
    else:
        assert df.values_0[0] == 3
        assert df.values_1[0] == 5

    # with set_metric_names()
    study.set_metric_names(["v0", "v1"])
    df = study.trials_dataframe(attrs=attrs, multi_index=multi_index)
    if multi_index:
        assert df.get("values")["v0"][0] == 3
        assert df.get("values")["v1"][0] == 5
    else:
        assert df.get(f"values_v0")[0] == 3
        assert df.get(f"values_v1")[0] == 5
```